Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,85 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
-                                Py_TYPE(self)->tp_name,
-                                GET_TD_DAYS(self));
+    PyObject *args = PyList_New(0), *days = NULL, *seconds = NULL, *microseconds = NULL;
+
+    if (args == NULL) {
+        return NULL;
+    }
+
+    if (GET_TD_DAYS(self) != 0) {
+        days = PyUnicode_FromFormat("days=%d", GET_TD_DAYS(self));
+        if (days == NULL) {
+            Py_DECREF(args);
+            return NULL;
+        }
+
+        if (PyList_Append(args, days) < 0) {
+            Py_DECREF(args);
+            Py_XDECREF(days);
+            return NULL;
+        }
+    }
```

Comment:
You can decref `days` here. This will simplify the cases for seconds and microseconds.